src/sentry/monitors/logic/mark_failed.py

```diff
@@ -37,23 +46,24 @@ def mark_failed(
 
 def mark_failed_threshold(
     failed_checkin: MonitorCheckIn,
-    ts: datetime | None,
+    ts: datetime,
     failure_issue_threshold: int,
 ):
     from sentry.signals import monitor_environment_failed
 
     monitor_env = failed_checkin.monitor_environment
 
-    # update monitor environment timestamps on every check-in
-    if ts is None:
-        next_checkin_base = timezone.now()
-        last_checkin = monitor_env.last_checkin or timezone.now()
-    else:
-        next_checkin_base = ts
-        last_checkin = ts
```

Comment:
The change here is that `last_checkin` is now stored as the `failed_checkin.date_added` instead of `timestamp.now()` for missed, or in the case of 'updated' check-ins, the update time.